FridgeExercise/Program.cs

```diff
@@ -1,2 +1,242 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿using FridgeExercise;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static FridgeExercise.Item;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace FridgeExercise
+{
+	
+	internal class Program
+	{
+		public void SortheRefrigerators(List<Refrigerator> AllRefrigerators)
+		{
+			AllRefrigerators.Sort((x, y) => y.SpaceIsLeftInTheFridge(y.Shelves).CompareTo(x.SpaceIsLeftInTheFridge(x.Shelves)));
+			//to work on a copied list do orderbydescending
+			foreach (Refrigerator refrigerator in AllRefrigerators)
+			{
+				Console.WriteLine(refrigerator.ToString(refrigerator));
+			}
+			
+		}
+		public void PrintAllInTheFridge(Refrigerator refrigerator)
+		{
+			Console.WriteLine(refrigerator.ToString(refrigerator));
+			foreach (Shelf shelf in refrigerator.Shelves)
+			{
+				Console.WriteLine(shelf.ToString(shelf));
+				foreach (Item item in shelf.Items)
+				{
+					Console.WriteLine(item.ToString(item));
+				}
+			}
+		}
+		public void CheckValidTypeItem(string KosherTypeToEat, string typeToEat,Refrigerator fridge)
+		{
+			if (typeToEat == "Food" || typeToEat == "Drink")
+			{
+				if (KosherTypeToEat == "Parve" || KosherTypeToEat == "Dairy" || KosherTypeToEat == "Meat")
+					fridge.WhatDoIWantToEat(typeToEat, KosherTypeToEat, fridge.Shelves);
+				else
+					Console.WriteLine("There is no such a type:" + typeToEat);
+			}
+			else
+				Console.WriteLine("There is no such a kosherType: " + KosherTypeToEat);
+
+		}
+		public void InsertItem(Refrigerator fridge)
+		{
+			int day = 0, month = 0, year = 0;
+			Console.WriteLine("Insert an Item:");
+			Console.WriteLine("Name:");
+			string name = Console.ReadLine();
+			Console.WriteLine("Expired date:in this tamplate:dd-mm-yyyy");
+			string expireDate = Console.ReadLine();
+			string[] dateSeperated = expireDate.Split('-');
+			if (dateSeperated.Count() == 3)
+			{
+				try
+				{
+					var checkDate = new DateTime(Int32.Parse(dateSeperated[2]), Int32.Parse(dateSeperated[1]), Int32.Parse(dateSeperated[0]));
+					day = Int32.Parse(dateSeperated[0]);
+					month = Int32.Parse(dateSeperated[1]);
+					year = Int32.Parse(dateSeperated[2]);
+				}
+				catch
+				{
+					Console.WriteLine("The date is not correct");
+					return;
+				}
+
+			}
+			Console.WriteLine("How much room it takes in square meters ?:enter just a double number");
```

Comment:
```suggestion
			Console.WriteLine("How much space does it require in square meters?");
```

No need for the double part - you could convert whatever number you're given to a double

Reply:
Another thing to keep in mind regarding strings, especially the ones that aren't dynamic - if they are printed often, you may want to store them in a constant field. It means they would get created only once, and whenever you'd like to use them, you could just reference their fields.
I'm not going to require that of you, but it is something I suggest you keep in mind going forward.